sotodlib/toast/ops/load_context.py

```diff
@@ -0,0 +1,711 @@
+        group_firstobs = groupdist[comm.group].offset
+        group_numobs = groupdist[comm.group].n_elem
+
+        # Every group loads its observations
+        for obindx in range(group_firstobs, group_firstobs + group_numobs):
+            obs_name = obs_props[obindx]["name"]
+
+            # One process in the group loads the metadata, builds the focalplane
+            # model, and broadcasts to the rest of the group.
+
+            det_props = None
+            obs_meta = None
+            n_samp = None
+            rate = None
+            if comm.group_rank == 0:
+                meta = self.context.get_meta(obs_name, dets=det_select)
+                # Read telescope data to get number of samples
+                axtemp = self.context.get_obs(obs_name, no_signal=True)
+                n_samp = len(axtemp[self.ax_times])
+                # Now that we have timestamps loaded, update our focalplane sample rate
+                (rate, dt, dt_min, dt_max, dt_std) = toast.utils.rate_from_times(
+                    axtemp[self.ax_times]
+                )
+                # For each element of meta we do the following:
+                # - If the object has one axis and it is the detector axis,
+                #   treat it as a column in the detector property table
+                # - If the object has one axis and it is the sample axis,
+                #   it will be loaded later as shared data.
+                # - If the object has multiple axes, load it later
+                # - If the object has no axes, then treat it as observation
+                #   metadata
+                # - If the object is a nested AxisManager, descend and apply
+                #   the same steps as above.
+                obs_meta = dict()
+                fp_cols = dict()
+                self._parse_meta(meta, None, obs_meta, None, fp_cols)
+
+                # Construct table
+                det_props = QTable(fp_cols)
+
+            if comm.comm_group is not None:
+                obs_meta = comm.comm_group.bcast(obs_meta, root=0)
+                det_props = comm.comm_group.bcast(det_props, root=0)
+                n_samp = comm.comm_group.bcast(n_samp, root=0)
+                rate = comm.comm_group.bcast(rate, root=0)
+            rate = u.Quantity(rate, u.Hz)
+
+            # Create the observation.  We intentionally use the generic focalplane
+            # and class here, in case we are loading data from legacy experiments.
+
+            # Convert any focalplane quaternion offsets to toast format
+
+            name_col = Column(name="name", data=det_props["det_info_readout_id"])
+
+            if "focal_plane_xi" in det_props.colnames:
+                quat_data = toast.instrument_coords.xieta_to_quat(
+                    det_props["focal_plane_xi"],
+                    det_props["focal_plane_eta"],
+                    det_props["focal_plane_gamma"],
+                )
+            else:
+                # No detector offsets yet
```

Comment:
I'm a little bit conflicted with defaulting to unit quaternions here. Perhaps we should require the user to indicate if they want detector quaternions or not and then we'll know if their absence is an error?

Reply:
Only some toast operators require the detector pointing offsets.  If the user was loading data without those offsets, then presumably they would be intending to either not use them or to run code to estimate / populate them.  In either case, we could simply exclude them from the focalplane table.  Then any attempted use would raise an exception.  I'll make that change.

Reply:
On further inspection, this change would require a small upstream in the `toast.Focalplane` class.  I'm going to leave the null quats here for now, but will fix after that change upstream.